test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+use serde_json::Value;
+use serde::{Deserialize, Deserializer};
+use up_rust::{
+    Data, UAttributes, UAuthority, UCode, UEntity, UMessage, UMessageType, UPayload,
+    UPayloadFormat, UPriority, UResource, UUri, UUID,
+};
+
+use protobuf::{MessageField, SpecialFields};
+
+pub fn convert_json_to_jsonstring<T: serde::Serialize>(value: &T) -> String {
+    serde_json::to_string(value).expect("Failed to convert to JSON string")
+}
+
+#[derive(Debug, Default)]
+pub struct WrapperUUri(pub UUri);
```

Comment:
You could consider factoring this out into a PR into `up-rust` so that you don't have to have a wrapper type and implement serialization and deserialization here. Might be a good idea.

Reply:
sagar opened an issue in up-rust